packages/angular/src/generators/scam/lib/create-module.ts

```diff
@@ -31,6 +32,28 @@ export function createScam(tree: Tree, schema: Schema) {
         projectConfig.projectType === 'application' ? 'app' : 'lib',
         componentNames.fileName
       );
+
+  if (schema.path) {
+    const folderToPlaceIn = schema.path.split(
+      joinPathFragments(
+        projectConfig.sourceRoot,
+        projectConfig.projectType === 'application' ? 'app' : 'lib'
+      )
+    )[1];
+    componentDirectory = schema.flat
+      ? joinPathFragments(
+          projectConfig.sourceRoot,
+          projectConfig.projectType === 'application' ? 'app' : 'lib',
+          folderToPlaceIn
+        )
+      : joinPathFragments(
+          projectConfig.sourceRoot,
+          projectConfig.projectType === 'application' ? 'app' : 'lib',
+          folderToPlaceIn,
+          componentNames.fileName
+        );
```

Comment:
We shouldn't need to do this if we already have validated the provided path starts with the `sourceRoot` (though we should change it to `root`). We should just use the provided path as the folder where to place the component.

I think this whole block can be simplified to:

```typescript
componentDirectory = schema.flat
  ? normalizePath(schema.path)
  : joinPathFragments(schema.path, componentNames.fileName);
```

> **Note**: `normalizePath` can be imported from `@nrwl/devkit`.

Reply:
I feel like i overthought this for some reason and. I'm not sure why